lib/ansible/template/__init__.py

```diff
@@ -540,6 +540,8 @@ def is_template(self, data):
                 new = self.do_template(data, fail_on_undefined=True)
             except (AnsibleUndefinedVariable, UndefinedError):
                 return True
+            except AnsibleFilterError as e:
+                return 'Mandatory variable not defined.' == to_text(e)
```

Comment:
What happens if it is another filter that caused an error and not `mandatory`?

Reply:
bad things ... need to narrow it down